tests/test_loss.py

```diff
@@ -152,3 +153,32 @@ def test_logsumexp():
     b = np.full(n, 10000, dtype='float64')
     desired = 10000.0 + np.log(n)
     assert_almost_equal(_logsumexp(b), desired)
+
+
+def test_baseline_predictions():
+    # Tests for the get_baseline_prediction() method of the losses
+    rng = np.random.RandomState(0)
+
+    # check that least squares return the mean of y_true
+    loss = _LOSSES['least_squares']()
+    y_true = rng.normal(size=100)
+    assert_almost_equal(loss.get_baseline_prediction(y_true, None),
+                        y_true.mean())
+
+    # test when there is only one class in training data
+    for loss_name, n_trees_per_iteration in (
+            ('binary_crossentropy', 1), ('categorical_crossentropy', 5)):
+        for y_true in (np.zeros(shape=100), np.ones(shape=100)):
+            y_true = y_true.astype(np.float32)
+            loss = _LOSSES[loss_name]()
+            init = loss.get_baseline_prediction(y_true, n_trees_per_iteration)
+            assert_all_finite(init)
+
+    # check that the init values are constants
+    for loss_name, n_trees_per_iteration in (
+            ('binary_crossentropy', 1), ('categorical_crossentropy', 5)):
+        loss = _LOSSES[loss_name]()
+        y_true = rng.randint(0, n_trees_per_iteration + 1,
+                             size=1000).astype(np.float32)
+        init = loss.get_baseline_prediction(y_true, n_trees_per_iteration)
+        assert np.unique(init).size == n_trees_per_iteration
```

Comment:
Could you please split this test in 3 tests: one for each loss and actually test that the baseline prediction matches the class frequency ratios for the classification losses.

Please also add assertions on the expected shape of `loss.get_baseline_prediction(y_train, n_trees_per_iteration)` in each case.

Reply:
>test that the baseline prediction matches the class frequency ratios for the classification losses.

You mean check that the values are exactly as expected (log(1 / 1-p))? I'm not sure how to do that without simply copy pasting the code from the loss, but then this test doesn't check much.

Reply:
Yes, consider this simple test as a non regression test and a way to clarify what is the intent of baseline prediction.

Reply:
Or even consider this test as fine grained specification/documentation from the API consumer point of view.